### ISSUE TITLE
Performance of interp_like when converting dfsu to dfs2 for large files 

### ISSUE THREAD

**Issue by ryan-kipawa** [DHI/mikeio#413]
**Is your feature request related to a problem? Please describe.**
Converting large dfsu files to dfs2 seems to take a long time (for a 120mb item, it took roughly 6 hours). I used the following code:

```
ds = mikeio.read(dfsu_filename, items=item, time=time)
da = ds[item]
g = da.geometry.get_overset_grid(dx=dxdy)
da_grid = da.interp_like(g)
```

The PC has 64GB memory and it's CPU is: Intel Xeon E5-2650 v4 @ 2.20Ghz.

**Describe the solution you'd like**
Is there anything I could do to make this run faster?

**Describe alternatives you've considered**
I haven't tried anything yet. One idea is to vectorize the for loop in "_interp_itemstep".

**Additional context**

**Comment by jsmariegaard** [comment]
You could also try with 

    da_grid = da.interp_like(g, n_nearest=1)

which use nearest point instead of IDW interpolation.

**Comment by jsmariegaard** [comment]
Hmm, yes, the contains() method (to find out if points are inside the mesh) is actually what takes the most time in this case. But closely followed by the dot product. It may be better to set extrapolate=True and then remove points outside the domain afterwards

    > da_grid = da.interp_like(g, extrapolation=True)   # will not call contains()

**Comment by cethceth** [comment]
I have a similar size dfsu file and have the same issue, but I get an error when I try to set 'extrapolation=True':
![image](https://user-images.githubusercontent.com/117161494/199225072-7954c8db-bb52-418b-ad90-deb28a551cb3.png)
Any idea how to solve this?

**Comment by cethceth** [comment]
.. and can it really be true that this process takes that long in Python when in Mike Zero it is quite fast?

**Comment by jsmariegaard** [comment]
@cethceth, which version of MIKE IO are you using?

**Comment by cethceth** [comment]
Solved it with 'extrapolate' instead of 'extrapolation' :)

**Comment by ecomodeller** [comment]
> .. and can it really be true that this process takes that long in Python when in Mike Zero it is quite fast?

Are you referring to Create new Grid series - From Dfsu file (with the less than intuitive grid specification)?
![image](https://user-images.githubusercontent.com/614215/199556594-568e354d-5e58-488e-af6e-8dee41a2e137.png)

**Comment by cethceth** [comment]
> 

Yes. But it works fine with extrapolate now. Any idea of how to mask the output grid (fast)? I would prefer not having to load shapefiles.

**Comment by ecomodeller** [comment]
@cethceth maybe you can tell us a bit more about your use-case., not sure we can do anything about it right now, but it is useful to know. I suppose converting the dfsu to dfs2 is simply an intermediate step?

**Comment by cethceth** [comment]
@ecomodeller we want an automatic tool for converting dfsu files to tiff files which we use for reporting, calculation of damage cost etc.. We have that now. But since the output tiff files are not masked this has to be done as a post processing step in GIS. I was hoping to gather all the processing in python.. and also avoid having to load shapefiles in python for masking the output. If there was a way to get the extent of the dfsu file which could then be used when masking the dfs2/tiff file in python that would be really nice :)
